### PR TITLE
Fix HomeAssistant capability and add Features

### PR DESCRIPTION
**Pull request by kingy444**
Maybe you want to bump further like 2.12? 

Just given the changes they were not insignifigant. I only bumped to 2.11.2 for testing with HomeAssistant

The most signifigant changes made were:
- managing the session entirely within init (and not within update_status) 
- removing the async with for client_session in update_status as this was closing the home assistant entity
- Error catching on the TaskGroup
- move from retry to tenacity (retry was not working and project appears stale)

This also supports filter clean / dirty and only returns zones that are enabled on the controller rather than including the zones that are disabled aswell

### CODE REVIEW

File: .bumpversion.cfg [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [bumpversion]
-current_version = 2.11.1
+current_version = 2.11.2
```
Comment (fredrike) [inline .bumpversion.cfg:2]: We create versions manually, please revert.

File: setup.py [hunk 2]
```diff
@@ -7,7 +7,7 @@
 
 setup(
     name='pydaikin',
-    version='2.11.1',
+    version='2.11.2',
```
Comment (fredrike) [inline setup.py:10]: See above.

```suggestion
    version='2.11.1',
```

File: .bumpversion.cfg (hunk 1 above)
Comment (fredrike) [reply .bumpversion.cfg:2]:
```suggestion
current_version = 2.11.1
```

File: requirements.txt [hunk 3]
```diff
@@ -1,4 +1,4 @@
 netifaces
 aiohttp
 urllib3
-retry
\ No newline at end of file
+tenacity
```
Comment (fredrike) [inline requirements.txt:4]: tenacity looks like an awesome tool!

Make sure it is added to the pre-commit file too.

Can we use it for other parts in the code?

**Review by fredrike — CHANGES_REQUESTED** [review]
Some minor changes, you are right this looks like it qualifies a minor version-bump.

**Comment by kingy444** [comment]
I think that may have been sorted with the move to DaikinFactory class instead of the Appliance.factory method linked in the issue description 

I'm definitely not seeing the same long boot times in my dev environment so I'm hopeful it would

File: requirements.txt (hunk 3 above)
Comment (kingy444) [reply requirements.txt:4]: sorry completly missed these comments. I have added to the pre-commit

Comment (kingy444) [reply requirements.txt:4]: I havent delved into the code for what else it might be useful for, but there are indeed some cool features there

**Comment by kingy444** [comment]
I have fixed most the pre-commit issues - but i think the pylint is failing due to this pre-commit https://github.com/fredrike/pydaikin/pull/6/commits/5973258fa4c9b0b3618b53ea835be6edc057a680

**Comment by fredrike** [comment]
The details is outlined here: https://results.[pre-commit.ci](https://results.pre-commit.ci/run/github/757419802/1716528114.uHUD88kaR5eVG68JIn2aiA)/run/github/757419802/1716528114.uHUD88kaR5eVG68JIn2aiA

There is something with pylint and netifaces, I can take a look at that.

File: pydaikin/discovery.py [hunk 4]
```diff
@@ -37,11 +37,13 @@ def poll(self, stop_if_found=None, ip=None):  # pylint: disable=invalid-name
             broadcast_ips = [ip]
         else:
             # get all IPv4 definitions in the system
+            # pylint: disable=c-extension-no-member
```
Comment (fredrike) [inline pydaikin/discovery.py:40]: This is needed for pylint.

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -29,7 +29,7 @@ repos:
         - aiohttp==3.7.3
         - netifaces==0.11.0
         - urllib3==1.26.3
-        - retry==0.9.2
+        - tenacity==8.2.3
         exclude: 'tests/'
         args:
         - --ignore=setup.py
```
Comment (fredrike) [inline .pre-commit-config.yaml:35]:
```suggestion
        - --ignore=setup.py
        - --extension-pkg-allow-list=netifaces
```

File: pydaikin/daikin_airbase.py [hunk 6]
```diff
@@ -59,7 +59,7 @@ def parse_response(response_body):
 
     def __init__(
```
Comment (fredrike) [inline pydaikin/daikin_airbase.py:60]:
```suggestion
    def __init__(  # pylint:disable=useless-parent-delegation
```

File: pydaikin/daikin_airbase.py [hunk 7]
```diff
@@ -59,7 +59,7 @@ def parse_response(response_body):
...
         self, device_id, session=None
-    ):  # pylint:disable=useless-super-delegation
+    ) -> None:  # pylint:disable=useless-parent-delegation
```
Comment (fredrike) [inline pydaikin/daikin_airbase.py:62]:
```suggestion
    ) -> None:
```

File: pydaikin/discovery.py (hunk 4 above)
Comment (fredrike) [inline pydaikin/discovery.py:40]:
```suggestion
```

File: pydaikin/discovery.py [hunk 8]
```diff
@@ -37,11 +37,13 @@ def poll(self, stop_if_found=None, ip=None):  # pylint: disable=invalid-name
...
             net_groups = [
                 netifaces.ifaddresses(i)[netifaces.AF_INET]
                 for i in netifaces.interfaces()
                 if netifaces.AF_INET in netifaces.ifaddresses(i)
             ]
+            # pylint: enable=c-extension-no-member
```
Comment (fredrike) [inline pydaikin/discovery.py:46]:
```suggestion
```

Comment (fredrike) [reply pydaikin/discovery.py:46]: Remove extra pylint rows.

**Review by fredrike — CHANGES_REQUESTED** [review]

**Comment by fredrike** [comment]
Thanks a lot @kingy444 !

New version published to https://pypi.org/project/pydaikin/2.12.0/

**Pull request merged**